src/main/fc/fc_msp.c

```diff
@@ -1892,6 +1893,148 @@ static mspResult_e mspFcProcessInCommand(uint16_t cmdMSP, sbuf_t *src)
     return MSP_RESULT_ACK;
 }
 
+static const setting_t *mspReadSettingName(sbuf_t *src)
+{
+    char name[SETTING_MAX_NAME_LENGTH];
+    uint8_t c;
+    size_t s = 0;
+    while (1) {
+        if (!sbufReadU8Safe(&c, src)) {
+            return NULL;
+        }
+        name[s++] = c;
+        if (c == '\0') {
+            break;
+        }
+        if (s == SETTING_MAX_NAME_LENGTH) {
+            // Name is too long
+            return NULL;
+        }
+    }
+    return setting_find(name);
+}
+
+static bool mspSettingCommand(sbuf_t *dst, sbuf_t *src)
+{
+    const setting_t *setting = mspReadSettingName(src);
+    if (!setting) {
+        return false;
+    }
+
+    const void *ptr = setting_get_value_pointer(setting);
+
+    switch (SETTING_TYPE(setting)) {
+        case VAR_UINT8:
+            FALLTHROUGH;
+        case VAR_INT8:
+            sbufWriteU8(dst, *((uint8_t*)ptr));
+            break;
+        case VAR_UINT16:
+            FALLTHROUGH;
+            break;
```

Comment:
Really break here, not just fall through ?

Reply:
@stronnag Nice catch!